### PR TITLE
*: Introduce diagnostics API

### PR DESCRIPTION
**Pull request by rleungx**
### What problem does this PR solve?

Issue Number: Close #xxx

### What is changed and how does it work?

```commit-message
```

### Check List

Tests

- Unit test
- Integration test
- Manual test (add detailed scripts or steps below)
- No code

Code changes

- Has configuration change
- Has HTTP API interfaces changed (Don't forget to [add the declarative for the new API](https://github.com/tikv/pd/blob/master/docs/development.md#updating-api-documentation))
- Has persistent data change

Side effects

- Possible performance regression
- Increased code complexity
- Breaking backward compatibility

Related changes

- PR to update [`pingcap/docs`](https://github.com/pingcap/docs)/[`pingcap/docs-cn`](https://github.com/pingcap/docs-cn):
- PR to update [`pingcap/tiup`](https://github.com/pingcap/tiup):
- Need to cherry-pick to the release branch

### Release note

```release-note
Please add a release note.

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

If you don't think this PR needs a release note then fill it with None.
```

### CODE REVIEW

File: server/api/diagnostic.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+	}
+}
+
+func (h *diagnosticHandler) GetDiagnosticResult(w http.ResponseWriter, r *http.Request) {
+	var input map[string]interface{}
+	if err := apiutil.ReadJSONRespondError(h.rd, w, r.Body, &input); err != nil {
+		return
+	}
+
+	name, ok := input["name"].(string)
```
Comment (CabinfeverB) [inline server/api/diagnostic.go:44]: in `router.go`, param `name` is in URL. we should choose one way

File: server/cluster/diagnostic_manager.go [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+	}
+	queue := d.dryRunResults[name]
+	queue.Put(result.Timestamp, result)
+	return result, nil
+}
+
+// TODO: implement this function for each scheduler maybe.
+func (d *diagnosticManager) analyze(name string, ops []*operator.Operator, plans []plan.Plan, ts uint64) *DiagnosticResult {
+	res := &DiagnosticResult{Name: schedulers.BalanceRegionName, Timestamp: ts, Status: normal}
+	if isPaused, _ := d.cluster.IsSchedulerPaused(name); isPaused {
```
Comment (CabinfeverB) [inline server/cluster/diagnostic_manager.go:70]: It seems we don't need to dry run if isPaused is true. Can we put `DryRun` in `analyze `?

File: server/cluster/diagnostic_manager.go [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+		res.Status = paused
+		return res
+	}
+	if isDisabled, _ := d.cluster.IsSchedulerDisabled(name); isDisabled {
```
Comment (CabinfeverB) [inline server/cluster/diagnostic_manager.go:74]: diito

**Comment by rleungx** [comment]
It has been done via https://github.com/tikv/pd/pull/5471.

**Pull request closed without merging**